docs/configuration/http_conn_man/header_sanitizing.rst

```diff
@@ -0,0 +1,34 @@
+.. _config_http_conn_man_header_sanitizing:
+
+HTTP header sanitizing
+======================
+
+For security reasons, Envoy will "sanitize" various incoming HTTP headers depending on whether the
+request is an internal or external request. The sanitizing action depends on the header and may
+result in addition, removal, or modification. Ultimately, whether the request is considered internal
+or external is governed by the :ref:`x-forwaded-for <config_http_conn_man_headers_x-forwarded-for>`
+header (please read the linked section carefully as how Envoy populates the header is complex and
+depends on the :ref:`use_remote_address <config_http_conn_man_use_remote_address>` setting).
+
+Envoy will potentially sanitize the following headers:
+
+* :ref:`x-forwarded-client-cert <config_http_conn_man_headers_x-forwarded-client-cert>`
+* :ref:`x-forwarded-for <config_http_conn_man_headers_x-forwarded-for>`
+* :ref:`x-forwarded-proto <config_http_conn_man_headers_x-forwarded-proto>`
+* :ref:`x-request-id <config_http_conn_man_headers_x-request-id>`
+* :ref:`x-envoy-internal <config_http_conn_man_headers_x-envoy-internal>`
+* :ref:`x-envoy-force-trace <config_http_conn_man_headers_x-envoy-force-trace>`
+* :ref:`x-envoy-external-address <config_http_conn_man_headers_x-envoy-external-address>`
+* :ref:`x-envoy-downstream-service-node <config_http_conn_man_headers_downstream-service-node>`
+* :ref:`x-envoy-downstream-service-cluster
+  <config_http_conn_man_headers_downstream-service-cluster>`
+* :ref:`x-envoy-retry-on <config_http_filters_router_x-envoy-retry-on>`
+* :ref:`x-envoy-grpc-retry-on <config_http_filters_router_x-envoy-grpc-retry-on>`
+* :ref:`x-envoy-max-retries <config_http_filters_router_x-envoy-max-retries>`
+* :ref:`x-envoy-upstream-alt-stat-name <config_http_filters_router_x-envoy-upstream-alt-stat-name>`
+* :ref:`x-envoy-upstream-rq-timeout-ms <config_http_filters_router_x-envoy-upstream-rq-timeout-ms>`
+* :ref:`x-envoy-upstream-rq-per-try-timeout-ms
```

Comment:
Random thought that might not be practical; would it make sense to have distinct header namespace for headers that are to be filtered on external req? I.e. you could just strip `x-envoy-req-*`.

Reply:
I would like a better way of grouping things that need to be sanitized, but not sure it's reasonable to change these things now.